setup-build-env.sh

```diff
@@ -4,57 +4,56 @@ ensure_lxd() {
   if ! command -v lxc version &> /dev/null
   then
     echo "LXD could not be found. Please ensure LXD exists."
-    exit
+    exit 1
   fi
 }
 
 build_image_in_container() {
   
-  local OS_NAME="${OS_NAME:-ubuntu}"
-  local OS_VERSION="${OS_VERSION:-22.04}"
-  local ARCH="${ARCH:-$(uname -m)}"
   local IMAGE_ALIAS="${IMAGE_ALIAS:-${OS_NAME}-${OS_VERSION}-${ARCH}}"
 
-  local BUILD_PREREQS_PATH="${BUILD_PREREQS_PATH:-/home/ubuntu/remote-build/files}"
-  local DOTNET_SDK="${DOTNET_SDK:-dotnet-sdk-7.0.100-linux-ppc64le.tar.gz}"
-  local PATCH_FILE="${PATCH_FILE:-runner-ppc64le.patch}"
+  local BUILD_PREREQS_PATH="${SRCDIR}/build-files"
+  if [ ! -d "${BUILD_PREREQS_PATH}" ]; then
+      echo "Check the BUILD_PREREQS_PATH specification" >&2
+      exit 3
+  fi
+  local PATCH_FILE="${PATCH_FILE:-runner-${ARCH}.patch}"
 
   local BUILD_CONTAINER
   BUILD_CONTAINER="gha-builder-$(date +%s)"
-  lxc launch "${OS_NAME}:${OS_VERSION}" "${BUILD_CONTAINER}" 
+  lxc launch "${LXD_CONTAINER}" "${BUILD_CONTAINER}" 
   lxc ls
   
   # give container some time to wake up
-  sleep 10
+  sleep 5
   
   echo "Copy the build-image script into gha-builder"
-  lxc file push "${BUILD_PREREQS_PATH}/build-image.sh"  "${BUILD_CONTAINER}/home/ubuntu/"
+  echo ${BUILD_PREREQS_PATH}
+  echo ${BUILD_PREREQS_PATH}/build-image.sh-${ARCH}
+  lxc file push --mode 0755 "${BUILD_PREREQS_PATH}/build-image.sh-${ARCH}" "${BUILD_CONTAINER}${BUILD_HOME}/build-image.sh"
+  sleep 10s
```

Comment:
Ditto on sleeps.  I would swear that they are high on this list when we look into testing failures as a class of problems because, mostly simply, "load varies".

Reply:
Again, interlock these with Anup.  I actually despise them for the reasons I already listed.  :)  I prefer pure deterministic whenever possible and those are assuming there is indeterminism.  If we can verify that there isn't, e.g. commands sync before next command starts, or a danged check-if-its-there-yet loop which isn't much better but at least doesn't lead to (as) random results...it would make me happier.  And I like being happy!  :)

Reply:
I agree, the sleep can be substituted for a wait-loop, I'll make the changes

Reply:
Do you want to send me the fix so we don't have multiple branches and the merge issues that will go with that?

Reply:
yeah that works, thanks @nealef

Reply:
Issue raised: https://github.com/anup-kodlekere/gaplib/issues/2